### PR TITLE
Add `scrollTo` and `select` to @ember/test-helpers

### PR DESCRIPTION
**Pull request by kwliou**
`scrollTo` and `select` were added in 2.0 https://github.com/emberjs/ember-test-helpers/blob/master/CHANGELOG.md
https://github.com/emberjs/ember-test-helpers/blob/master/addon-test-support/%40ember/test-helpers/index.ts

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by jamescdavis — CHANGES_REQUESTED** [review]
Hi @kwliou! Thanks for contributing! There's a few issues here. In addition to the comments I made inline, there should be tests for these added to https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ember__test-helpers/ember__test-helpers-tests.ts for these new functions.

File: types/ember__test-helpers/index.d.ts
Comment (kwliou) [reply types/ember__test-helpers/index.d.ts:1]: @jamescdavis  Ok I tried different permutations of paths and couldn't get it to work.
Can you help with the error `Expected directory ember__test-helpers/v1 to be the path mapping for ember__test-helpers`?

Comment (kwliou) [reply types/ember__test-helpers/index.d.ts:1]: Also is adding new functions considered a breaking change?

**Comment by chriskrycho** [comment]
Ugh, so sorry I missed this till now! Reviewing. 😓

**Comment by chriskrycho** [comment]
Oh, I see, @jamescdavis had done a review pass before. I will let him continue helping get it across the line!

File: types/ember__test-helpers/index.d.ts
Comment (jamescdavis) [reply types/ember__test-helpers/index.d.ts:1]: The issue I have with adding these methods and bumping the major version of these types without splitting off a v1 is that then we don't have a way to back port types fixes for folks using 1.x. I'll see if I can track down why it was failing.

**Comment by jamescdavis** [comment]
I'll see if I can figure out why things were breaking when adding v1.

**Comment by sandersn** [comment]
I didn't see the v1/ files (because of the force-pushes maybe?), but the error sounds like the tsconfig is misconfigured.

**Comment by andrewbranch** [comment]
@jamescdavis @kwliou I believe the issue you hit was fixed by https://github.com/microsoft/DefinitelyTyped-tools/pull/215. Can you give it another shot? Ping me if you still get errors. (Make sure you merge/rebase with master and `npm install` if testing locally.)

**Review by amcasey — CHANGES_REQUESTED** [review]
Waiting for sign-off from @jamescdavis.

**Comment by jamescdavis** [comment]
@kwliou can you try creating v1 types again now that the above issue has been resolved?

**Comment by kwliou** [comment]
@jamescdavis It's still the same errors from the first time after rebasing.

**Comment by jamescdavis** [comment]
@andrewbranch @sandersn any ideas?

File: types/ember__test-helpers/v1/tsconfig.json
Comment (amcasey) [inline types/ember__test-helpers/v1/tsconfig.json:20]: This is not my area of expertise, but I think you want `[ember__test-helpers/v1"]`

Comment (kwliou) [reply types/ember__test-helpers/v1/tsconfig.json:20]: @amcasey I tried it and I get
```
Older version 1 must have a "paths" entry of "ember__test-helpers/*": ["ember__test-helpers/v1/*"]
```

Comment (amcasey) [reply types/ember__test-helpers/v1/tsconfig.json:20]: What happens if you add that verbatim?

Comment (amcasey) [reply types/ember__test-helpers/v1/tsconfig.json:20]: I'm still mostly guessing, but what about
```json
            "@ember/*": ["ember__*"],
            "@ember/test-helpers": ["ember__test-helpers/v1"],
            "@ember/test-helpers/*": ["ember__test-helpers/v1/*"]
```

Seems like you still need the `@ember` path mapping.

Comment (amcasey) [reply types/ember__test-helpers/v1/tsconfig.json:20]: Once I do that, the v1 test fails because there are calls to `scrollTo` and `select`, which seem to be the new functions (i.e. not included in v1)?

Comment (kwliou) [reply types/ember__test-helpers/v1/tsconfig.json]: @amcasey I updated the PR and it still fails with the same error.

**Comment by amcasey** [comment]
The tests seem happy, so it must be some other part of the infrastructure that wants a specific format.  Ping @andrewbranch @sandersn

File: types/ember__test-helpers/v1/tsconfig.json
Comment (amcasey) [reply types/ember__test-helpers/v1/tsconfig.json]: Yep.  Sorry, it's over my head at this point.  The only other thing I can think of (if you can't wait for an expert to show up) is to give it exactly what it wants (in addition to the sensible path mappings you already have).

```json
            "@ember/*": ["ember__*"],
            "@ember/test-helpers": ["ember__test-helpers/v1"],
            "@ember/test-helpers/*": ["ember__test-helpers/v1/*"],
            "ember__test-helpers": ["ember__test-helpers/v1"],
            "ember__test-helpers/*": ["ember__test-helpers/v1/*"]
```

Seems like an infrastructure bug to me but, again, I don't know much about this functionality.

File: types/ember__test-helpers/v1/tsconfig.json
Comment (andrewbranch) [inline types/ember__test-helpers/v1/tsconfig.json:22]: It seems like CI wants this key to be mangled... but I don’t know why that wasn’t the case for this one: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ckeditor__ckeditor5-utils/v10/tsconfig.json#L15

Seems like we still have a bug... I’ll look into it today.

**Comment by andrewbranch** [comment]
New error, which hopefully means the one that didn’t make sense is fixed:

> Error: ember__test-helpers v1.7 has unused path mapping for @ember/test-helpers

**Comment by andrewbranch** [comment]
Sorry, didn’t meant to push that just yet. You all can just sit back and 🍿 while I fight CI here. Getting it green is on me.

**Review by dfreeman — APPROVED** [review]
I think @jamescdavis is on vacation, but this looks good to me now that the `v1` business is sorted out. Thanks for slogging through this, @kwliou!

**Review by amcasey — APPROVED** [review]

**Comment by kwliou** [comment]
Ready to merge

**Pull request merged**
